src/api/jobs/jobs.go

```diff
@@ -0,0 +1,46 @@
+package jobs
+
+import (
+	"log"
+	"time"
+)
+
+// Periodic is an interface for managing periodic job invocation
+type Periodic interface {
+	// Do begins the periodic job. It starts the first execution of the job, and then is
+	// responsible for executing it every Frequency() thereafter
+	Do() error
+	Frequency() time.Duration
+	Name() string
+}
+
+// DoPeriodic calls p.Do() once, and then again every p.Frequency() on each element p in pSlice.
+// For each p in pSlice, a new goroutine is started, and the returned channel can be closed
+// to stop all of the goroutines.
+func DoPeriodic(pSlice []Periodic) chan<- struct{} {
```

Comment:
also, might be good to abstract away the returned `chan<- struct{}` into a `func()`. user can then call it to close all the goroutines instead of dealing with the channel implementation detail